OpenXmlPowerTools/OpenXmlRegex.cs

```diff
@@ -64,17 +64,55 @@ public static int Match(IEnumerable<XElement> content, Regex regex, Action<XElem
          */
         public static int Replace(IEnumerable<XElement> content, Regex regex, string replacement, Func<XElement, Match, bool> doReplacement)
         {
-            return ReplaceInternal(content, regex, replacement, doReplacement, false, null, true);
+            Func<XElement, Match, string> callback;
+            if (doReplacement == null)
+            {
+                callback = (x, m) => replacement;
+            }
+            else
+            {
+                callback = (x, m) => doReplacement(x, m) ? replacement : null;
+            }
+            return ReplaceInternal(content, regex, callback, false, null, true);
+        }
+        /*
+         * Call doReplacement(elem, match) on every match.
+         * If it returns null, do nothing
+         * If it returns "", delete the matched text
+         * If it returns something else, replace the matched text with the new text
+         */
+        public static int Replace(IEnumerable<XElement> content, Regex regex, Func<XElement, Match, string> doReplacement)
+        {
+            return ReplaceInternal(content, regex, doReplacement, false, null, true);
         }
-
         /*
          * This overload enables not coalescing content, which is necessary for DocumentAssembler.
          */
         public static int Replace(IEnumerable<XElement> content, Regex regex, string replacement, Func<XElement, Match, bool> doReplacement, bool coalesceContent)
         {
-            return ReplaceInternal(content, regex, replacement, doReplacement, false, null, coalesceContent);
+            Func<XElement, Match, string> callback;
+            if (doReplacement == null)
+            {
+                callback = (x, m) => replacement;
+            }
+            else
+            {
+                callback = (x, m) => doReplacement(x, m) ? replacement : null;
+            }
```

Comment:
This if-else-block is repeated thrice in three overloads of `Replace`. Could it be factored out?